Feature(IconGeneration)

alaxion opened the pull request:
New feature: generate icon according to selected colors and add icon file to books.

alaxion committed:
feat(SelectAnnotation.ui): modify UI

alaxion committed:
Merge branch 'Feature-Annotation' of https://github.com/Ad147/Magic into Feature-Annotation

alaxion committed:
feat(SelectAnnotation): add function generate icon

alaxion committed:
feat(SelectAnnotation.ui): modify color select button style

alaxion committed:
feat(SelectAnnotation)

1. Function: automatically add icon file to epub
2. delete renderButtion()

alaxion committed:
feat(SelectAnnotation): add svg resource file to qrc

Commit:
feat(layoutengine): update engine lib

alaxion committed:
feat(addStylesheet): add stylesheet resource

alaxion committed:
feat(addStylesheet): add stylesheet for annotation icon

Commit:
fix(misc): misc

alaxion committed:
Merge pull request #3 from luojilab/Dedao

Dedao

alaxion committed:
feat(replace icon): replace file when icon changed

Lmonster commented on src/Dialogs/SelectAnnotation.cpp:
```diff
@@ -28,27 +28,54 @@ SOFTWARE.
+const QString S_srcBgColor = QString("#000000");
+const QString S_srcFgColor = QString("#FFFFFF");
+const QSize S_iconSize = QSize(100, 100);
+const QString S_defaultIconSrc = QString("../Images/AnnoIcon0.png");
```
这种初始化不建议，直接const QString S_defaultIconSrc = "" 。现在的这种构造函数调用调用了两次构造函数。上面的字符串构造函数都是这个问题

alaxion replied:
ok

Lmonster commented on src/Dialogs/SelectAnnotation.cpp:
```diff
@@ -68,12 +95,143 @@ int SelectAnnotation::insertAnnotation(const QString &annoText, const QString &a
+    
+    // Replace color old colors with new colors
+    if (S_lastBgColor != m_bgColor)
+    {
```
if 的左括号，跟if在一行

alaxion replied:
ok

Lmonster commented on src/Dialogs/SelectAnnotation.h:
```diff
@@ -44,44 +47,66 @@ class SelectAnnotation;
+    QString m_fgColor;
+    QImage m_iconImg;
+    QByteArray m_svgBytes;
+    QSvgRenderer m_renderer;
```
m_renderer没有必要声明为一个成员变量

alaxion committed:
fix(SelectAnnotation.h/.cpp)

1. fix curly brace format
2. fix const QString construct
3. modify rederer as a local variable

Lmonster commented on src/Dialogs/SelectAnnotation.cpp:
```diff
@@ -68,12 +95,131 @@ int SelectAnnotation::insertAnnotation(const QString &annoText, const QString &a
+            QMessageBox::warning(this, tr("Magic"), tr("添加样式文件失败。Adding CSS file failed. (AnnoErr5)"));
+            return;
+        }
+        S_annoStylesheetAdded = true;
```
为什么要用静态变量？
